app/controllers/single_reservations_controller.rb

```diff
@@ -0,0 +1,84 @@
+class SingleReservationsController < ApplicationController
```

Comment:
I'm not in love with the name--especially in the route/url--but I haven't come up with anything better.